src/components/Executions/useTaskExecutions.ts

```diff
@@ -1,38 +1,42 @@
+import { useAPIContext } from 'components/data/apiContext';
 import { every } from 'lodash';
 import {
     ExecutionData,
-    getTaskExecutionData,
     limits,
-    listTaskExecutions,
     NodeExecution,
     NodeExecutionIdentifier,
+    SortDirection,
     TaskExecution,
-    TaskExecutionIdentifier
+    TaskExecutionIdentifier,
+    taskSortFields
 } from 'models';
 import { useDataRefresher } from '../hooks';
 import { FetchableData } from '../hooks/types';
 import { useFetchableData } from '../hooks/useFetchableData';
 import { executionRefreshIntervalMs } from './constants';
 import { nodeExecutionIsTerminal, taskExecutionIsTerminal } from './utils';
 
-const doFetchTaskExecutions = async (id: NodeExecutionIdentifier) => {
-    const { entities } = await listTaskExecutions(id, {
-        limit: limits.NONE
-    });
-    return entities;
-};
-
 /** A hook for fetching the list of TaskExecutions associated with a
  * NodeExecution
  */
 export function useTaskExecutions(
     id: NodeExecutionIdentifier
 ): FetchableData<TaskExecution[]> {
+    const { listTaskExecutions } = useAPIContext();
     return useFetchableData<TaskExecution[], NodeExecutionIdentifier>(
         {
             debugName: 'TaskExecutions',
             defaultValue: [],
-            doFetch: doFetchTaskExecutions
+            doFetch: async (id: NodeExecutionIdentifier) => {
+                const { entities } = await listTaskExecutions(id, {
+                    limit: limits.NONE,
+                    sort: {
```

Comment:
Does this replace the sorting that was removed in the earlier file?

Reply:
It does. The earlier file was doing the sorting client-side. This uses the sorting built in to the API so that the results come back already in the desired order.